src/features/comments.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-param-reassign */
+import { PayloadAction, createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { createComment, deleteComment, getPostComments } from '../api/comments';
+import { Comment } from '../types/Comment';
+
+type State = {
+  comments: Comment[],
+  loaded: boolean,
+  hasError: boolean,
+  visible: boolean,
+  submitting: boolean,
+};
+
+const initialState: State = {
+  comments: [],
+  loaded: false,
+  hasError: false,
+  visible: false,
+  submitting: false,
+};
+
+export const initComments = createAsyncThunk('comments/fetch',
+  (postId: number) => {
+    return getPostComments(postId);
+  });
+
+export const addComment = createAsyncThunk('createComment/post',
+  (data: Omit<Comment, 'id'>) => {
+    const newComment = createComment(data);
+
+    return newComment;
+  });
+
+export const removeComment = createAsyncThunk('removeComment/delete',
```

Comment:
The naming convention for the action type in 'createAsyncThunk' is incorrect. You should follow the pattern `${sliceName}/${reducerName}` to maintain consistency and readability.